lib/src/io_client.dart

```diff
@@ -72,3 +73,16 @@ class IOClient extends BaseClient {
     _inner = null;
   }
 }
+
+Uri _responseUrl(Request request, HttpClientResponse response) {
+  var redirects = response.redirects;
+
+  if (redirects.isEmpty) {
+    return request.url;
+  }
+
+  var location = redirects.last.location;
```

Comment:
Actually I might just have to run through the array in case there was some ping ponging between servers. Not likely but certainly possible.